### PR TITLE
vim-patch:8.0.1553

### PR DESCRIPTION
**Pull request by lokeshmohanty**

### CODE REVIEW

**Comment by mhinz** [comment]
Okay, well, let's fix that branch first. ;-)

Please do this in your branch:

```
$ git remote update origin/master
$ git rebase origin/master
$ git rebase -i origin/master
<in all lines but the first change `pick` to `fixup`; then `:wq`>
<make sure that all commits were "squashed" into one single commit>
$ git push -f <the remote that points to git@github.com:lokesh1197/neovim>
```

This should clean up this PR by discarding all the commits that touched unrelated files.

File: runtime/doc/various.txt [hunk 1]
```diff
@@ -37,23 +37,34 @@ CTRL-L			Clear and redraw the screen.  The redraw may happen
 <
 :as[cii]	or					*ga* *:as* *:ascii*
 ga			Print the ascii value of the character under the
-			cursor in decimal, hexadecimal and octal.  For
-			example, when the cursor is on a 'R':
+			cursor in decimal, hexadecimal and octal.
+			Mnemonic: Get Ascii value.
+			
+			For example, when the cursor in on a 'R':
```
Comment (oni-link) [inline runtime/doc/various.txt]: in -> is

File: src/nvim/digraph.c [hunk 2]
```diff
@@ -1442,6 +1442,37 @@ int do_digraph(int c)
   return c;
 }
 
+/*
+ * Find a digraph for "val".  If found return the string to display it.
+ * If not found return NULL.
+ */
+    char_u *get_digraph_for_char(val)
+    int val;
```
Comment (oni-link) [inline src/nvim/digraph.c]: More up to date syntax should be used:  `char_u *get_digraph_for_char(int val) {`

File: src/nvim/digraph.c [hunk 3]
```diff
@@ -1442,6 +1442,37 @@ int do_digraph(int c)
...
+{
+    int		i;
+    int		use_defaults;
```
Comment (oni-link) [inline src/nvim/digraph.c]: Loop variables should be declared in the loop.

File: src/nvim/digraph.c [hunk 4]
```diff
@@ -1442,6 +1442,37 @@ int do_digraph(int c)
   return c;
 }
 
+/*
+ * Find a digraph for "val".  If found return the string to display it.
+ * If not found return NULL.
+ */
```
Comment (oni-link) [inline src/nvim/digraph.c]: Function comments need new style (`/// ...`), see other functions.

File: src/nvim/ex_cmds.c [hunk 5]
```diff
@@ -119,6 +119,9 @@ void do_ascii(exarg_T *eap)
   char buf1[20];
   char buf2[20];
   char_u buf3[7];
+#ifdef FEAT_DIGRAPHS
```
Comment (oni-link) [inline src/nvim/ex_cmds.c]: `FEAT_DIGRAPHS` should always be defined.

File: src/nvim/digraph.c (hunk 3 above)
Comment (lokesh1197) [reply src/nvim/digraph.c]: ok will do

**Comment by lokesh1197** [comment]
Can anyone please tell me whats going wrong here? Or how can I resolve these

**Comment by lokesh1197** [comment]
@justinmk  I have fixed the lint issues related to the files modified by me. But there are lint issues with other files that I haven't touched. Should I fix them as well?

**Comment by justinmk** [comment]
@lokesh1197 Please rebase, and fix any errors that `make lint` complains about in a separate commit named `lint`.

**Comment by lokesh1197** [comment]
@justinmk  Can you please tell me what I am supposed to do now?

**Comment by teto** [comment]
@lokesh1197 
The CI should be green, seems like some tests still fail, for instance if you click on the travis check you can see
```
xecuting Test_z_bug()
Executed 322 tests
1 FAILED:
Found errors in Test_ga_command():
function RunTheTest[35]..Test_ga_command line 4: Expected '\n<<01>>  1,  Hex 01,  Oct 001, Digr SH' but got '\n<<01>>  1,  Hex 01,  Oct 001,  Digr SH'
function RunTheTest[35]..Test_ga_command line 5: Expected '\n<<09>>  9,  Hex 09,  Oct 011, Digr HT' but got '\n<<09>>  9,  Hex 09,  Oct 011,  Digr HT'
function RunTheTest[35]..Test_ga_command line 9: Expected '\n<^A>  1,  Hex 01,  Oct 001, Digr SH' but got '\n<^A>  1,  Hex 01,  Oct 001,  Digr SH'
function RunTheTest[35]..Test_ga_command line 10: Expected '\n<^I>  9,  Hex 09,  Oct 011, Digr HT' but got '\n<^I>  9,  Hex 09,  Oct 011,  Digr HT'
SKIPPED Test_isnan(): Nvim does not support isnan()
SKIPPED Test_tag_file_encoding(): Nvim removed test83-tags2, test83-tags3

```

**Comment by lokesh1197** [comment]
Thanks @teto . I have made the necessary corrections

File: src/nvim/digraph.c [hunk 6]
```diff
@@ -1448,6 +1448,32 @@ int do_digraph(int c)
...
+  char_u *get_digraph_for_char(int val) {
+  digr_T *dp;
+  static char_u r[3];
+
+  for (int use_defaults = 0; use_defaults <= 1; use_defaults++) {
+    if (use_defaults == 0) {
+      dp = (digr_T *)user_digraphs.ga_data;
+    } else {
+      dp = digraphdefault;
+      for (int i = 0; use_defaults ? dp->char1 != NUL \
```
[the remark this answers is not in the document]
Comment (lokesh1197) [reply src/nvim/digraph.c]: Done.

**Comment by lokesh1197** [comment]
@teto Can you please tell me what is the error this time? I am unable to decipher the log of travis-ci

**Comment by justinmk** [comment]
Merged, thanks.

In the future it will save time for you and reviewers if you use `scripts/vim-patch.sh` (described in the [wiki](https://github.com/neovim/neovim/wiki/Merging-patches-from-upstream-Vim)).

**Pull request closed without merging**